File: backend/src/main/java/org/globe42/web/countries/CountryDTO.java
package org.globe42.web.countries;

import org.globe42.domain.Country;

/**
 * A DTO for the {@link Country} entity
 * @author JB Nizet
 */
public final class CountryDTO {
    private final String id;
    private final String name;

    public CountryDTO(Country country) {
        this.id = country.getId();
        this.name = country.getName();
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}

File: backend/src/main/kotlin/org/globe42/web/countries/CountryDTO.kt
package org.globe42.web.countries

import org.globe42.domain.Country

/**
 * A DTO for the [Country] entity
 * @author JB Nizet
 */
data class CountryDTO(val id: String, val name: String) {
    constructor(country: Country): this(country.id, country.name)
}
